pipeline/transformer/transformer_common/src/main/scala/weco/pipeline/transformer/identifiers/IdentifierRegexes.scala

```diff
@@ -0,0 +1,63 @@
+package weco.pipeline.transformer.identifiers
+
+import scala.util.matching.Regex
+
+object IdentifierRegexes {
+  /*
+   * A Sierra system number (aka a bnumber) is composed of:
+   * - A prefix 'b'
+   * - A 7 digit identifier
+   * - A check character, which can be any digit or an 'x'
+   */
+  lazy val sierraSystemNumber: Regex = "^b[0-9]{7}[0-9x]$".r
+
+  /*
+   * A Sierra identifier is 7 digits
+   * (the same as the middle portion of a system number)
+   */
+  lazy val sierraIdentifier: Regex = "^[0-9]{7}$".r
+
+  /* Regex for parsing a Miro ID.  Examples of valid IDs, all taken from
+   * the Sierra data:
+   *
+   *    L0046161, V0033167F1, V0032544ECL
+   *
+   * Regarding suffixes (from a slack discussion with collections):
+   *
+   * You should find two types of suffix, one beginning with E and one beginning with F.
+   * The E suffix was used to introduce a further character that distinguished two or
+   * more works that were photographed on the same piece of film
+   * (colour transparency and black&white negative).
+   *
+   *   L = Left
+   *   R = Right.
+   *   T=Top
+   *   B=Bottom
+   *   TR =Top Right
+   *   etc.
+   *
+   * To add to that, there are also M series images with suffix EA, EB, EC, ED, EE, and EF.
+   * They don't seem to mean anything (like L = left), they seem to just be running letters
+   * to slip the image in line.
+   */
+  lazy val miroImageNumber: Regex =
+    "^[A-Z][0-9]{7}[A-Z]{0,3}[0-9]?$".r
```

Comment:
I would double-check this – things like `AS0000081FH23` are real Miro IDs, just very weird ones.

Reply:
Ha, in that case I don't think we'd ever have been able to merge them as this came from the Sierra transformer. I'll make it so that the suffix-of-the-suffix can be any number of digits?

Reply:
A good compromise here might be to look at Miro IDs that are currently cleared for the catalogue API, because those are the things we can care about matching.

(I realise I said "just add validation" and have then left you the fun of actually dealing with the hard parts of it.)

Reply:
Of the Miro IDs that are currently cleared for the catalogue API, these 5 do not match the regex:

```
V0009118ETLL
L0025088F01
V0009118EBLL
V0009118ETRR
V0009118EBRR
```

So I'm not going to overthink it and will change the regex to:

```
^[A-Z][0-9]{7}[A-Z]{0,4}[0-9]{0,2}$
```